logger/logger.go

```diff
@@ -166,11 +166,18 @@ func (l *Log) NoGuardDebug(msg string) {
 	l.log.Output(3, "DEBUG: "+msg)
 }
 
-// New creates a log.Logger using the given io.Writer and flag.
-func New(w io.Writer, flag int) (Logger, error) {
+type LoggerOptions struct {
+	// forceDebug can be set if we want debug traces even if not directly
+	// enabled by environment or kernel command line.
+	ForceDebug bool
+}
+
+// New creates a log.Logger using the given io.Writer and flag, using the
+// options from opts.
+func New(w io.Writer, flag int, opts *LoggerOptions) (Logger, error) {
```

Comment:
bit of a nitpick but if we could add a nil check at the top, like:
```suggestion
func New(w io.Writer, flag int, opts *LoggerOptions) (Logger, error) {
  if opts == nil {
    opts = &LoggerOptions{}
  }
```

we could simplify all the calls without opts by passing nil. It would also be consistent with how we usually deal with opts I think

Reply:
Right, very good suggestion. In the end I changed to what Miguel suggested as it is what we do in most cases.